File: lib/limelight/commands/open_command.rb
#- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

require 'limelight/commands/command'

module Limelight
  module Commands

    # See the following usage summary.
    #
    #    Usage: limelight open <production_path>
    #        Open a limelight production.
    #        options:
    #        -h, --help                       Prints this usage summary.
    #        -d, --drb_port=<port>            Publish productions using DRb starting with specified port.
    #
    class OpenCommand < Command

      install_as "open"

      DEFAULT_PRODUCTION = File.expand_path($LIMELIGHT_HOME + "/productions/playbills.lll")

      attr_reader :drb_port

      def initialize
        self.print_backtrace = true
      end

      def self.description
        return "Open a limelight production."
      end

      protected ###########################################

      def parameter_description
        return "<production_path>"
      end

      def do_requires       
        Main.initialize_context
        require 'limelight/producer'
      end

      def build_options(spec) #:nodoc:
        spec.on("-d <port>", "--drb_port=<port>", "Publish productions using DRb starting with specified port.") { |value| @drb_port = value }
      end

      def parse_remainder(args)
        @production_path = args.shift || DEFAULT_PRODUCTION
      end

      def process
        Context.instance.studio.publish_productions_on_drb(@drb_port.to_i) if @drb_port
        Context.instance.studio.open(@production_path)
      end

    end
    
  end
end

File: src/limelight/commands/OpenCommand.java
//- Copyright © 2008-2011 8th Light, Inc. All Rights Reserved.
//- Limelight and all included source files are distributed under terms of the MIT License.

package limelight.commands;

import limelight.Boot;
import limelight.Context;

import java.util.Map;

public class OpenCommand extends Command
{
  private static Arguments arguments;
  private boolean shouldBoot = true;

  public static Arguments arguments()
  {
    if(arguments == null)
    {
      arguments = new Arguments();
      arguments.addSwitchOption("r", "rigger", "Open the 'Rigger' scene too");
      arguments.addOptionalParameter("production", "Path to production directory, .llp file, or .lll file.  If none is provided, the Playbills production is opened.");
    }
    return arguments;
  }

  @Override
  public void doExecute(Map<String, String> args)
  {
    String production = args.get("production");
    if(production == null)
      production = defaultProduction();

    if(shouldBoot)
      Boot.boot();

    if(args.containsKey("rigger"))
      Context.instance().studio.utilitiesProduction().openRigger();

    Context.instance().studio.open(production);
  }

  @Override
  public String description()
  {
    return "Opens a Limelight Production";
  }

  @Override
  public String name()
  {
    return "open";
  }

  @Override
  public Arguments getArguments()
  {
    return arguments();
  }

  public String defaultProduction()
  {
    return Context.fs().join(Context.instance().limelightHome, "productions", "playbills.lll");
  }

  public void setShouldBoot(boolean value)
  {
    shouldBoot = value;
  }
}
